src/Sentry/Laravel/Tracing/Storage/TracingFilesystem.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace Sentry\Laravel\Tracing\Storage;
+
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Sentry\Breadcrumb;
+use Sentry\Laravel\Integration;
+use Sentry\Laravel\Util\Filesize;
+use Sentry\Tracing\SpanContext;
+use function Sentry\trace;
+
+/**
+ * Decorates the underlying filesystem by wrapping all calls to it with tracing.
+ *
+ * Parameters such as paths, directories or options are attached to the span as data,
+ * parameters that contain file contents are omitted due to potential problems with
+ * payload size or sensitive data.
+ */
+class TracingFilesystem implements Filesystem
+{
+    /** @var Filesystem */
+    protected $filesystem;
+
+    /** @var array */
+    protected $defaultData;
+
+    /** @var bool */
+    protected $recordSpans;
+
+    /** @var bool */
+    protected $recordBreadcrumbs;
+
+    public function __construct(Filesystem $filesystem, array $defaultData, bool $recordSpans, bool $recordBreadcrumbs)
+    {
+        $this->filesystem = $filesystem;
+        $this->defaultData = $defaultData;
+        $this->recordSpans = $recordSpans;
+        $this->recordBreadcrumbs = $recordBreadcrumbs;
+    }
+
+    /**
+     * @param list<mixed> $args
+     * @param array<string, mixed> $data
+     */
+    protected function withTracing(string $method, array $args, string $description, array $data)
+    {
+        $op = "file.{$method}"; // See https://develop.sentry.dev/sdk/performance/span-operations/#web-server
+        $data = array_merge($data, $this->defaultData);
+
+        if ($this->recordBreadcrumbs) {
```

Comment:
Are breadcrumbs also considered tracing?

Reply:
Perhaps the class and method should be renamed to something like `SentryFilesystem` and `withSentry` then.

Reply:
You might be right yeah